SmartDeviceLink/SDLIAPSession.m

```diff
@@ -139,6 +273,51 @@ - (SDLStreamErrorHandler)streamErroredHandler {
     };
 }
 
+- (SDLStreamHasSpaceHandler)streamHasSpaceHandler {
+    __weak typeof(self) weakSelf = self;
+    
+    return ^(NSStream *stream) {
+        __strong typeof(weakSelf) strongSelf = weakSelf;
+        
+        if (strongSelf.isDataSession){
+            SessionSendHandler sendCompletion;
+            @synchronized (strongSelf) {
+                sendCompletion = strongSelf.sendBlock;
+            }
+            
+            if (sendCompletion != nil){
+                NSError *err = nil;
+                BOOL sendComplete = sendCompletion(&err);
+                if (err != nil){
+                    NSLog(@"ERROR!!! Output stream reported %@", err);
```

Comment:
Thanks for the clarification Joel. I could see cases where the head unit is busy, say indexing the iTunes media library on the iOS device or updating the SXM guide. Could we accomplish the same result - reliable reconnection - by making the delay shorter and allowing retries, but within a background task if the app is backgrounded?